src/Statuses/Status.ts

```diff
@@ -22,15 +22,15 @@ export class Status {
      * @type {Status}
      * @memberof Status
      */
-    public static DONE: Status = Status.makeDone();
+    public static DONE: Status = new Status(new StatusConfiguration('x', 'Done', ' ', true, StatusType.DONE));
```

Comment:
I believe that this change stops the PR from being a pure refactor.

Previously, the same instance was returned for every use of `DONE`. and `makeDone()` created new instances.

I am pretty sure that there were tests that specifically depended on this difference in behaviour.

If they pass now, I worry that it means we have incomplete test coverage.

Reply:
Before doing the refactor, I called `Status.TODO` and `Status.makeTodo()` and compared them with `expect().toEqual()` and `expect().toBe()`. I found that all of them passed so I continued with this.

Reply:
> I am pretty sure that there were tests that specifically depended on this difference in behaviour.
> 
> If they pass now, I worry that it means we have incomplete test coverage.

Can it be that the tests are still there but the behaviour part somewhere in the code is gone?

Reply:
> Before doing the refactor, I called `Status.TODO` and `Status.makeTodo()` and compared them with `expect().toEqual()` and `expect().toBe()`. I found that all of them passed so I continued with this.

That's odd...

Here's what I find, with the current code on `main`... Note the `failing` one...

```ts
describe('Status Lifetime', () => {
    it('Status.DONE - toBe', () => {
        // Passes, as two successive uses of Status.DONE use the same object
        const done1 = Status.DONE;
        const done2 = Status.DONE;
        expect(done1).toBe(done2);
    });

    it('Status.DONE - toEqual', () => {
        const done1 = Status.DONE;
        const done2 = Status.DONE;
        expect(done1).toEqual(done2);
    });

    it.failing('Status.makeDone() - toEqual', () => {
        // Fails, as two successive calls to Status.makeDone() create two different objects
        const done1 = Status.makeDone();
        const done2 = Status.makeDone();
        expect(done1).toBe(done2);
    });

    it('Status.makeDone() - toEqual', () => {
        const done1 = Status.makeDone();
        const done2 = Status.makeDone();
        expect(done1).toEqual(done2);
    });
})
```

Reply:
The above was run with 9b428be5764f8d8c83e8c29096734563b85326e5

Reply:
Ok, I understand the behavior from the snippet you sent, but I winder what is the impact in the use of status as same object or as singleton. Do you remember in which tests was it an important factor?

Reply:
> Ok, I understand the behavior from the snippet you sent, but I winder what is the impact in the use of status as same object or as singleton. Do you remember in which tests was it an important factor?

No I don't.

So let's look at what changed...

The files in src/ that changed are:

- `src/Config/StatusSettings.ts`
- `src/Statuses/StatusRegistry.ts`

So can you follow through the uses of the changed code, to work out whether it matters that all calls to the change code now use the same instances, where previously they used unique instances.

If you are confident of that behavioural change, then I'm happy to go ahead and merge the PR.

(Asking as I'm not sufficiently pained by the existing behaviour to want to spend time figuring out the risk - sorry!)

Reply:
Sure, I did some digging and found:

- Bug #1535 is not reproducible on my branch (Just the first thing I found and wanted to be sure)
- PR #1455 that is related to nested objects inside `Status` objects, and that was fixed on the underlying `StatusConfiguration` layer
  - > I have found equality tests on Status objects obtained from the StatusRegistry were failing confusingly because Status objects returned from StatusRegistry had a duplicate nested configuration field, and I didn't understand why

Can it be that the issues you were referring to are issues with `StatusConfiguration` (that were fixed) and not `Status`? =)

I don't have any other proof of absence of change in behaviour besides existing tests. Also when I was refactoring, I tried breaking the code by supplying wrong statuses in `Status` statics and successfully broke many tests. Given that I don't have an absolute confidence in the coverage (as you mentioned before `If they (tests) pass now, I worry that it means we have incomplete test coverage.`), I can't guarantee that things won't go south from here. What I can guarantee is that I will be willing to spend my time to fix them if they do =) Maybe it will be an opportunity to discover where our coverage is failing =)

Reply:
> What I can guarantee is that I will be willing to spend my time to fix them if they do =) Maybe it will be an opportunity to discover where our coverage is failing =)

Ok, thank you.

The benefit of this PR is it removes the ambiguity or confusion about the differences between `Status.DONE` and `Status.makeDone()` etc...

So I will approve it and merge.